src/customRules/structuredImportsRule.ts

```diff
@@ -0,0 +1,94 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    'ruleName': 'structured-imports',
+    'description': 'Enforce structure to your imports. Import structure should be listed in the folloing order: modules, absolute imports,  relative parent/ancestor directories, relative sibling directors.',
+    'hasFix': false,
+    'optionsDescription': 'Not configurable.',
+    'options': null,
+    'optionExamples': null,
+    'type': 'style',
+    'typescriptOnly': false,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static STRUCTURED_IMPORTS_ABS_FIRST_ERROR = 'Imports should be listed in the following order: module imports, absolute imports, relative imports.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const structuredImportsWalker = new StructuredImportsWalker(sourceFile, this.getOptions());
+    return this.applyWithWalker(structuredImportsWalker);
+  }
+}
+
+enum ImportType {
+  Module = 0, // 'module'
+  Absolute = 1, // '/some/folder/file'
+  RelativeAncestor = 2, // '../parentFolder'
+  RelativeSibling = 3, // './siblingFolder'
+};
+
+const importStuctureOrder = [ImportType.Module, ImportType.Absolute, ImportType.RelativeAncestor, ImportType.RelativeSibling];
+
+class StructuredImportsWalker extends Lint.RuleWalker {
+  private previousImport: ImportType;
+  private currentImport: ImportType;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+  }
+
+  // e.g. 'import Foo from './foo';'
+  public visitImportDeclaration(node: ts.ImportDeclaration) {
```

Comment:
Thanks for the feedback!
I couldn't run my tests with this, earlier today. It needed a bit more refactoring but it was a great starting point. 
Did you build then run the tests? I think that might of done it, it gets me all the time too.